checkout/access-checkout-android/src/main/java/com/worldpay/access/checkout/api/configuration/CardConfigurationParser.kt

```diff
@@ -65,19 +61,12 @@ internal class CardConfigurationParser : Deserializer<CardConfiguration>() {
                     IMAGES_FIELD
                 )
                 val brandImages = parseBrandImages(images)
-                val cvv = fetchOptionalObject(brandRoot,
-                    BRANDED_CVV_FIELD
-                )
-                val cvvConfig = cvv?.let { cvvRule -> parseCardValidationRule(cvvRule) }
-                val pans = fetchArray(brandRoot,
-                    PANS_FIELD
-                )
-                val panList = mutableListOf<CardValidationRule>()
-                for (panRuleIndex in 0 until pans.length()) {
-                    val panValidationRule = parseCardValidationRule(pans.getJSONObject(panRuleIndex))
-                    panList.add(panValidationRule)
-                }
-                brandsList.add(CardBrand(name, brandImages, cvvConfig, panList))
+                val cvv = toOptionalIntProperty(brandRoot, BRANDED_CVV_FIELD)
+                val cvvConfig = cvv?.let { cvvLength -> CardValidationRule(validLengths = listOf(cvvLength)) }
+                val pans = fetchOptionalArray(brandRoot, PANS_FIELD)
+                val matcher = toOptionalStringProperty(brandRoot, MATCHER_FIELD)
+                val panRule = CardValidationRule(matcher, parseLengths(pans as JSONArray))
```

Comment:
can we add some new lines in between these statements where it makes sense, it is quite hard to read

Reply:
upadted